src/klr.hosting.shared/RuntimeBootstrapper.cs

```diff
@@ -300,7 +300,7 @@ private static string[] ResolveSearchPaths(IEnumerable<string> libPaths, List<st
                     extension.Equals(".exe", StringComparison.OrdinalIgnoreCase))
                 {
                     // Add the directory to the list of search paths
-                    searchPaths.Add(Path.GetDirectoryName(application));
```

Comment:
Since `RuntimeBootstrapper` already does the following expansion:
`klr /path/to/MyApplication.dll` --->  `klr --lib /path/to/ MyApplication`.
So we only need to generate `--appbase /path/to/` part in this case.
Keeping logic here helps to reduce C++ code.